### PR TITLE
refactor environmental sequence

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

**Comment by djfr** [comment]
This approach works given the current group that is making decisions on what the regional map looks like.

If this was done properly and folks gave regions some thought, the mapping would have a lot more variations: CI 1 region, TEST 2 regions, PRE-PROD 4-6 regions, SAND 1 region and PROD 4-6 regions and in this case a more detailed mapping structure would be better then a simple if(environment is CI).

**Comment by VaclavK** [comment]
>           This approach works given the current group that is making decisions on what the regional map looks like.
> If this was done properly and folks gave regions some thought, the mapping would have a lot more variations: CI 1 region, TEST 2 regions, PRE-PROD 4-6 regions, SAND 1 region and PROD 4-6 regions and in this case a more detailed mapping structure would be better then a simple if(environment is CI).

yes indeed... but given current set up I think this is more readable
if any realignment happens, the package would have to be updated anyway (since we do not externalise this map nor plan to)

**Comment by djfr** [comment]
Looks good 👍

File: src/Eshopworld.DevOps/EswDevOpsSdk.cs [hunk 1]
```diff
@@ -103,23 +103,64 @@ public static string GetEnvironmentName()
...
-                return new DeploymentContext { PreferredRegions = new[] { Regions.WestEurope.ToRegionName() } };
+            if (string.IsNullOrWhiteSpace(value))
+                throw new ArgumentException("Null or empty value", nameof(value));
 
-            var region = GetEnvironmentVariable(DeploymentRegionEnvVariable);
+            foreach (var field in typeof(DeploymentRegion).GetFields().Where(fi=> !fi.IsSpecialName))
+            {
+                var regionDescriptor = (RegionDescriptorAttribute) field.GetCustomAttributes(
+                    typeof(RegionDescriptorAttribute),
+                    false).FirstOrDefault(); //but it will be there (see tests)
```
Comment (arturek) [inline src/Eshopworld.DevOps/EswDevOpsSdk.cs]: I think it's better to use .First if the collection is not expected to ever be empty. The comment contradicts what is in the code (I mean .FirstOrDefault and the later condition both suggest that this collection might be empty).

**Pull request merged**